src/sources/azure_blob/queue.rs

```diff
@@ -127,45 +143,88 @@ struct AzureStorageEvent {
     pub event_type: String,
 }
 
+#[derive(Debug, Snafu)]
+pub enum ProcessingError {
+    #[snafu(display("Could not decode Queue message with id {}: {}", message_id, error))]
+    InvalidQueueMessage {
+        error: serde_json::Error,
+        message_id: String,
+    },
+
+    #[snafu(
+        display("Container name of message and container client doesn't match, container: {}, message {}",
+            container,
+            message)
+    )]
+    ContainerNameDoesntMatch {
+        container: String,
+        message: String,
+    },
+
+    #[snafu(display("Failed to base64 decode message: {}", error))]
+    FailedDecodingMessageBase64 {
+        error: base64::DecodeError
+    },
+
+    #[snafu(display("Failed to utf8 decode message: {}", error))]
+    FailedDecodingUTF8 {
+        error: std::string::FromUtf8Error
+    },
+
+    #[snafu(display("Failed to get blob: {}", error))]
+    FailedToGetBlob {
+        error: azure_core::Error
+    },
+
+    #[snafu(display("Failed to parse {} as subject", subject))]
+    FailedToParseSubject {
+        subject: String,
+    }
+}
+
 async fn proccess_event_grid_message(
     message: Message,
     container_client: &ContainerClient,
     queue_client: &QueueClient,
     bytes_received: Registered<BytesReceived>,
-) -> Option<BlobPack> {
+) -> Result<Option<BlobPack>, ProcessingError> {
     let decoded_bytes = match BASE64_STANDARD.decode(&message.message_text) {
         Ok(decoded) => decoded,
         Err(e) => {
-            error!("Failed decoding base64: {}", e);
-            return None;
+            return Err(ProcessingError::FailedDecodingMessageBase64 { error: e });
```

Comment:
If I'm not mistaken, given we now have results here, all those match statements could be more concise:
```
    BASE64_STANDARD.decode(&message.message_text)
        .map_err(|e| ProcessingError::FailedDecodingMessageBase64 { error: e })?;
```